File: exercise-resources/py/crypto_2.py
from flask import Flask, jsonify, request
import bitcoin, hashlib, binascii, base58
import eth_keys, binascii
import json

app =Flask(__name__)
@app.route('/crypto2/eth_sign', methods=["POST"])
def eth_sign():
    values = request.get_json()
    if not values:
        return "Missing body", 400

    required = ["skey", "msg"]
    if not all(k in values for k in required):
        return "Missing values", 400

    # TODO: Not Implemented Yet
	
    response = {"signature": "TODO",
                "msg": "TODO"}

    return json.dumps(response), 201

@app.route('/crypto2/eth_sign_to_addr', methods=["POST"])
def eth_sign_to_addr():
    values = request.get_json()
    if not values:
        return "Missing body", 400

    required = ["signature", "msg"]
    if not all(k in values for k in required):
        return "Missing values", 400

    # TODO: Not Implemented Yet
	
    response = {"address": "TODO"}

    return json.dumps(response), 201

@app.route('/crypto2/eth_sign_verify', methods=["POST"])
def eth_sign_verify():
    values = request.get_json()
    if not values:
        return "Missing body", 400

    required = ["address", "signature", "msg"]
    if not all(k in values for k in required):
        return "Missing values", 400

    # TODO: Not Implemented Yet
	
    response = {"valid": "TODO"}
	
    return json.dumps(response), 201

@app.route('/crypto2/btc_skey_to_addr', methods=["POST"])
def btc_skey_to_addr():
    values = request.get_json()
    if not values:
        return "Missing body", 400

    required = ["skey"]
    if not all(k in values for k in required):
        return "Missing values", 400

    # TODO: Not Implemented Yet
	
    response = {"address": "TODO"}
	
    return json.dumps(response), 201

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=5000, debug=True)

File: exercise-resources/js/crypto_2.js
'use strict';
var express = require('express');
var bodyParser = require('body-parser');
var eth_crypto = require('eth-crypto');
let bitcoinjs = require('bitcoinjs-lib');
var http_port = 5000

function containsAll(body, requiredKeys) {
    return requiredKeys.every(elem => body.indexOf(elem) > -1) &&
           body.length == requiredKeys.length;
}

var initHttpServer = () => {
    var app = express();
    app.use(bodyParser.json());

    app.post('/crypto2/eth_sign', (req, res) => {
        var values = req.body
        if (Object.keys(values).length === 0) {
            return res.status(400).send('Missing Body')
        }

        var required = ["skey", "msg"]
        if (!containsAll(Object.keys(values), required)) {
            return res.status(400).send('Missing values')
        }

		// TODO: Not Implemented Yet

        res.send({ signature: "TODO", msg: "TODO"});
    });

    app.post('/crypto2/eth_sign_to_addr', (req, res) => {
        var values = req.body
        if (Object.keys(values).length === 0) {
            return res.status(400).send('Missing Body')
        }

        var required = ["signature", "msg"]
        if (!containsAll(Object.keys(values), required)) {
            return res.status(400).send('Missing values')
        }

		// TODO: Not Implemented Yet

        res.send({address: "TODO"});
    });

    app.post('/crypto2/eth_sign_verify', (req, res) => {
        var values = req.body
        if (Object.keys(values).length === 0) {
            return res.status(400).send('Missing Body')
        }

        var required = ["address", "msg", "signature"]
        if (!containsAll(Object.keys(values), required)) {
            return res.status(400).send('Missing values')
        }

        // TODO: Not Implemented Yet

        res.send({valid: "TODO"});
    });

    app.post('/crypto2/btc_skey_to_addr', (req, res) => {
        var values = req.body
        if (Object.keys(values).length === 0) {
            return res.status(400).send('Missing Body')
        }

        var required = ["skey"]
        if (!containsAll(Object.keys(values), required)) {
            return res.status(400).send('Missing values')
        }

        // TODO: Not Implemented Yet

        res.send({address: "TODO"});
    });

    app.listen(http_port, () => console.log("Listening http port: " + http_port));
}

initHttpServer();
